java/code/src/com/redhat/rhn/common/db/datasource/xml/oval_queries.xml

```diff
@@ -29,4 +29,10 @@
               AND cve.name = :cve_name;
         </query>
     </mode>
+
+    <mode name="check_oval_availability">
+        <query params="cpe">
+            SELECT 1 FROM suseOVALPlatform plat WHERE starts_with(:cpe, plat.cpe);
```

Comment:
Can you point me to what are the expected values on to be passed and the ones existing on the table. The order of arguments in this function looks to be on inverted order.

Reply:
The param cpe is the CPE of the system to scan. It is extracted as a salt grain and stored in the rhnServer table. You can find examples of it [here](https://github.com/chef/os_release). The plat.cpe is the CPE that we extract from OVAL files. 

We want to check if the CPE of the system corresponds to any CPEs we extracted from OVAL files. We could do that by searching for exact equality but that wouldn't match if the system has a CPE like `cpe:/o:redhat:enterprise_linux:8.0:GA` considering that this is the standard list of vulnerable RedHat 8 CPEs found in OVAL files:

```xml
   <affected_cpe_list>
    <cpe>cpe:/a:redhat:enterprise_linux:8</cpe>
    <cpe>cpe:/a:redhat:enterprise_linux:8::appstream</cpe>
    <cpe>cpe:/a:redhat:enterprise_linux:8::crb</cpe>
    <cpe>cpe:/a:redhat:enterprise_linux:8::highavailability</cpe>
    <cpe>cpe:/a:redhat:enterprise_linux:8::nfv</cpe>
    <cpe>cpe:/a:redhat:enterprise_linux:8::realtime</cpe>
    <cpe>cpe:/a:redhat:enterprise_linux:8::resilientstorage</cpe>
    <cpe>cpe:/a:redhat:enterprise_linux:8::sap</cpe>
    <cpe>cpe:/a:redhat:enterprise_linux:8::sap_hana</cpe>
    <cpe>cpe:/a:redhat:enterprise_linux:8::supplementary</cpe>
    <cpe>cpe:/o:redhat:enterprise_linux:8</cpe>
    <cpe>cpe:/o:redhat:enterprise_linux:8::baseos</cpe>
   </affected_cpe_list>
```

So we check for prefixes.